### ISSUE TITLE
Document and implement a design for images with multiple versions

### ISSUE THREAD

**Issue by stevehipwell** [chainguard-images/meta#19]
Apologies if this has been covered elsewhere but I don't see and logic for handling multiple versions of components such as the JDK, Node.js, .NET (when it arrives), or Ruby (when it arrives), etc? All of these components have overlapping LTS lifecycles where it's highly likely that multiple versions are going to be needed at the same time.

I've used all of the following patterns at some point and I don't think there is a categoric right answer, but IMHO there needs to be a consistent and well thought out one.

- Use a repo per image version e.g. `jdk17` & `jdk11`
  - Simplest solution
- Use a primary repo e.g. `jdk` and fork for LTS support e.g. `jdk` -> `jdk11`
  - Consistent with current behaviour
- Use single repo for all versions e.g. `jdk`
  - All logic in one place

**Comment by amouat** [comment]
Yeah, this is a topic with a lot of discussion internally. It also affects how we package things on wolfi. And it gets more complicated when you consider images that are dependent on multiple components with potentially different versions (e.g. Maven, which requires Java).

**Comment by stevehipwell** [comment]
I assume which OS, Wolfi or Alpine, the image is based on is also a consideration in that multi-dimensional matrix?

For context I think this problem is the genesis of [Nix](https://nixos.org/) (the OS, the package manager and the language).

**Comment by amouat** [comment]
Yes, although it's more "musl vs glibc" than Alpine vs Wolfi. At the moment we have "latest" and "latest-glibc" tags, which isn't ideal.

**Comment by stevehipwell** [comment]
@amouat am I remembering correctly that there is an ambition to replace Alpine with Wolfi for `musl` images?

**Comment by amouat** [comment]
This is something that's being discussed :)

**Comment by stevehipwell** [comment]
@amouat should this be closed with the monorepo being up and running?

**Comment by amouat** [comment]
I think we still need document this clearly for everyone.
